common/workunit/workunit.cpp

```diff
@@ -8721,6 +8725,46 @@ void CLocalWorkUnit::setDebugValue(const char *propname, const char *value, bool
     }
 }
 
+static const char *summaryTypeName(SummaryType type)
+{
+    switch (type)
+    {
+    case SummaryType::ReadFile: return "ReadFile";
+    case SummaryType::ReadIndex: return "ReadIndex";
+    case SummaryType::WriteFile: return "WriteFile";
+    case SummaryType::WriteIndex: return "WriteIndex";
+    case SummaryType::SpillFile: return "SpillFile";
+    case SummaryType::Service: return "Service";
+    default:
+        throwUnexpected();
+    }
+};
+
+bool CLocalWorkUnit::getSummary(SummaryType type, SummaryMap &map) const
+{
+    VStringBuffer xpath("Summaries/%s", summaryTypeName(type));
+    CriticalBlock block(crit);
+    IPropertyTree *list = p->queryPropTree(xpath);
+    if (!list)
+        return false;
+    Owned<IPropertyTreeIterator> r = list->getElements("v");
```

Comment:
I wonder if it would be much better to store all the (key,value) pairs as a single serialized string value.  

It is likely to be more efficient to store and retrieve.  @jakesmith how do you think the performance/storage requirements would compare.